src/_pytest/python.py

```diff
@@ -1419,16 +1414,11 @@ def __init__(
 
     def _initrequest(self):
         self.funcargs = {}
-        if self._isyieldedfunction():
-            assert not hasattr(
-                self, "callspec"
-            ), "yielded functions (deprecated) cannot have funcargs"
-        else:
-            if hasattr(self, "callspec"):
-                callspec = self.callspec
-                assert not callspec.funcargs
-                if hasattr(callspec, "param"):
-                    self.param = callspec.param
+        if hasattr(self, "callspec"):
+            callspec = self.callspec
+            assert not callspec.funcargs
+            if hasattr(callspec, "param"):
+                self.param = callspec.param
```

Comment:
This line appears to not be covered.. I've added `assert 0, 2` before it.
It was not covered previously then also.  https://codecov.io/gh/pytest-dev/pytest/src/3fd2f43fb6b7bcd2eb342d0d7056381937d3d8db/src/_pytest/python.py#L1431